tests/test_lib_policy.py

```diff
@@ -1390,6 +1391,39 @@ class MockUser(object):
         delete_policy("setpin_pol")
         db_token.delete()
 
+    def test_32_get_allowed_attributes(self):
+
+        class MockUser(object):
+            login = 'login'
+            realm = 'realm'
+            resolver = 'resolver'
+
+        user = MockUser()
+        g = FakeFlaskG()
+        g.policy_object = PolicyClass()
+        g.audit_object = FakeAudit()
+        g.logged_in_user = {"role": "admin", "username": "admin", "realm": ""}
+
+        d = get_allowed_custom_attributes(g, user)
+        self.assertEqual({"set": {}, "delete": []}, d)
+
+        set_policy("custom_attr", scope=SCOPE.ADMIN,
+                   action="{0!s}=:hello: one two".format(ACTION.SET_USER_ATTRIBUTES))
+        set_policy("custom_attr2", scope=SCOPE.ADMIN,
+                   action="{0!s}=:hello2: * :hello: three".format(ACTION.SET_USER_ATTRIBUTES))
+        set_policy("custom_attr3", scope=SCOPE.ADMIN,
+                   action="{0!s}=:*: on off".format(ACTION.SET_USER_ATTRIBUTES))
+        set_policy("custom_attr4", scope=SCOPE.ADMIN,
+                   action="{0!s}=*".format(ACTION.DELETE_USER_ATTRIBUTES))
+        g.policy_object = PolicyClass()
+
+        d = get_allowed_custom_attributes(g, user)
+        self.assertEqual(["*"], d.get("delete"))
+        self.assertEqual({"*": ["on", "off"],
+                          "hello": ["one", "two", "three"],
+                          "hello2": ["*"]}, d.get("set"))
```

Comment:
We need to keep an eye on this because lists aren't guaranteed to keep the order.

Reply:
:-) I also stumbled upon this one, two three in python 2.7.